unitary/examples/quantum_chinese_chess/move.py

```diff
@@ -217,3 +220,418 @@ def effect(self, *objects) -> Iterator[cirq.Operation]:
+                # We measure the ancilla to dertermine whether the slide could be made.
+                could_capture = world.pop([capture_ancilla])[0]
+            if not could_capture:
+                print(
+                    "Slide move not applied: either the source turns out be empty, or the path turns out to be blocked."
+                )
+                return iter(())
+            # Apply the capture.
+            # Force measure the source to be there.
+            world.force_measurement(source_0, 1)
+            source_0.is_entangled = False
+            # Target qubit is unhooked, i.e. it's replaced with a new ancilla with value = 0.
+            world.unhook(target_0)
+            # Now that it's qubit is reset, we also reset its classical properties.
+            target_0.reset()
+            alpha.PhasedMove()(source_0, target_0)
+            # Move the classical properties of the source piece to the target piece.
+            target_0.reset(source_0)
+            source_0.reset()
+            # Force measure the whole path to be empty.
+            for path_piece in quantum_path_pieces_0:
+                world.force_measurement(path_piece, 0)
+                path_piece.reset()
+            return iter(())
+        # For BASIC or EXCLUDED cases.
+        # Note that we DON'T require that the source piece is there.
+        source_0.is_entangled = True
+        conditions = [0] * len(quantum_path_pieces_0)
+        # We will apply the slide only if all quantum path pieces are empty.
+        alpha.quantum_if(*quantum_path_pieces_0).equals(*conditions).apply(
+            alpha.PhasedMove()
+        )(source_0, target_0)
+        # Copy the classical properties of the source piece to the target piece.
+        target_0.reset(source_0)
+        # Note that we should not reset source_0 (to be empty) since there are cases where
+        # the source is not moved. We'll later use board.update_board_by_sampling() to determine
+        # if the source piece needs to be reset to be empty.
+        return iter(())
+
+
+class SplitSlide(QuantumEffect):
+    """SplitSlide from source_0 to target_0 and target_1, with quantum_path_pieces_0 being the
+    quantum path pieces from source_0 to target_0, and quantum_path_pieces_1 being the quantum
+    path pieces from source_0 to target_1. The only accepted (default) move_variant is
+    - BASIC
+
+    Types of pieces that could make SplitSlide moves are: CANNON, ROOK, HORSE, and ELEPHANT.
+    """
+
+    def __init__(
+        self,
+        quantum_path_pieces_0: List[str],
+        quantum_path_pieces_1: List[str],
+    ):
+        self.quantum_path_pieces_0 = quantum_path_pieces_0
+        self.quantum_path_pieces_1 = quantum_path_pieces_1
+
+    def num_dimension(self) -> Optional[int]:
+        return 2
+
+    def num_objects(self) -> Optional[int]:
+        return 3
+
+    def effect(self, *objects) -> Iterator[cirq.Operation]:
+        source_0, target_0, target_1 = objects
```

Comment:
same pattern of use of objects as above, and the type annotation

Reply:
See reply above. Thanks.